PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'account.move'
-    _inherit = 'event.booth'
-    _inherit = 'event.booth.category'
-    _inherit = 'event.type.booth'
-    _inherit = 'product.product'
-    _inherit = 'product.template'
-    _inherit = 'sale.order'
-    _inherit = 'sale.order.line'

Total Changes: 19

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/event_booth_sale/models/account_move.py
+++ b/addons/event_booth_sale/models/account_move.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class AccountMove(models.Model):
-    _inherit = 'account.move'
+    _inherit = ['account.move']
 
     def _invoice_paid_hook(self):
         """ When an invoice linked to a sales order selling registrations is

--- a/addons/event_booth_sale/models/event_booth.py
+++ b/addons/event_booth_sale/models/event_booth.py
@@ -6,7 +6,7 @@ from odoo.exceptions import UserError
 
 
 class EventBooth(models.Model):
-    _inherit = 'event.booth'
+    _inherit = ['event.booth']
 
     # registrations
     event_booth_registration_ids = fields.One2many('event.booth.registration', 'event_booth_id')

--- a/addons/event_booth_sale/models/event_booth_category.py
+++ b/addons/event_booth_sale/models/event_booth_category.py
@@ -10,7 +10,7 @@ _logger = logging.getLogger(__name__)
 
 
 class EventBoothCategory(models.Model):
-    _inherit = 'event.booth.category'
+    _inherit = ['event.booth.category']
 
     def _default_product_id(self):
         return self.env.ref('event_booth_sale.product_product_event_booth', raise_if_not_found=False)

--- a/addons/event_booth_sale/models/event_booth_registration.py
+++ b/addons/event_booth_sale/models/event_booth_registration.py
@@ -9,7 +9,6 @@ class EventBoothRegistration(models.Model):
     """event.booth.registrations are used to allow multiple partners to book the same booth.
     Whenever a partner has paid their registration all the others linked to the booth will be deleted."""
 
-    _name = 'event.booth.registration'
     _description = 'Event Booth Registration'
 
     sale_order_line_id = fields.Many2one('sale.order.line', string='Sale Order Line', required=True, ondelete='cascade')

--- a/addons/event_booth_sale/models/event_type_booth.py
+++ b/addons/event_booth_sale/models/event_type_booth.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class EventTypeBooth(models.Model):
-    _inherit = 'event.type.booth'
+    _inherit = ['event.type.booth']
 
     product_id = fields.Many2one(related='booth_category_id.product_id')
     price = fields.Float(related='booth_category_id.price', store=True)

--- a/addons/event_booth_sale/models/product_product.py
+++ b/addons/event_booth_sale/models/product_product.py
@@ -1,8 +1,8 @@
 from odoo import api, models
 
 
-class Product(models.Model):
-    _inherit = 'product.product'
+class ProductProduct(models.Model):
+    _inherit = ['product.product']
 
     @api.onchange('service_tracking')
     def _onchange_type_event_booth(self):

--- a/addons/event_booth_sale/models/product_template.py
+++ b/addons/event_booth_sale/models/product_template.py
@@ -2,7 +2,7 @@ from odoo import _, api, fields, models
 
 
 class ProductTemplate(models.Model):
-    _inherit = 'product.template'
+    _inherit = ['product.template']
 
     service_tracking = fields.Selection(selection_add=[
         ('event_booth', 'Event Booth'),

--- a/addons/event_booth_sale/models/sale_order.py
+++ b/addons/event_booth_sale/models/sale_order.py
@@ -7,7 +7,7 @@ from odoo.osv import expression
 
 
 class SaleOrder(models.Model):
-    _inherit = 'sale.order'
+    _inherit = ['sale.order']
 
     event_booth_ids = fields.One2many('event.booth', 'sale_order_id', string='Booths')
     event_booth_count = fields.Integer(string='Booth Count', compute='_compute_event_booth_count')

--- a/addons/event_booth_sale/models/sale_order_line.py
+++ b/addons/event_booth_sale/models/sale_order_line.py
@@ -6,7 +6,7 @@ from odoo.exceptions import ValidationError
 
 
 class SaleOrderLine(models.Model):
-    _inherit = 'sale.order.line'
+    _inherit = ['sale.order.line']
 
     event_booth_category_id = fields.Many2one('event.booth.category', string='Booths Category', ondelete='set null')
     event_booth_pending_ids = fields.Many2many(
